src/main/java/nz/cri/gns/NZSHM22/opensha/inversion/MFDManipulation.java

```diff
@@ -0,0 +1,110 @@
+package nz.cri.gns.NZSHM22.opensha.inversion;
+
+import com.google.common.base.Preconditions;
+import org.opensha.commons.data.function.EvenlyDiscretizedFunc;
+import org.opensha.commons.data.uncertainty.UncertainIncrMagFreqDist;
+import org.opensha.sha.magdist.IncrementalMagFreqDist;
+
+import java.awt.geom.Point2D;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MFDManipulation {
+
+    /**
+     * This method returns the input MFD constraint
+     * restricted between minMag and maxMag. WARNING! This doesn't interpolate. For
+     * best results, set minMag & maxMag to points along original MFD constraint
+     * (i.e. 7.05, 7.15, etc)
+     *
+     * Can handle UncertainIncrMagFreqDist objects.
+     **/
+    public static IncrementalMagFreqDist restrictMFDConstraintMagRange(
```

Comment:
@voj as discussed we don't need this fn to support uncertainty - factor away this :)